lib/snowpack/type_cache.ex

```diff
@@ -4,16 +4,20 @@ defmodule Snowpack.TypeCache do
   """
 
   @cache :type_cache
+  @cache_limit 1_000
+
+  GenServer
+  @spec child_spec(any) :: Supervisor.child_spec()
+  def child_spec(opts) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, [opts]}
+    }
+  end
 
-  @spec start_link :: Supervisor.on_start()
-  def start_link do
-    case Mentat.start_link(name: @cache) do
-      {:ok, pid} ->
-        {:ok, pid}
-
-      {:error, {:already_started, pid}} ->
-        {:ok, pid}
-    end
+  @spec start_link(any()) :: Supervisor.on_start()
+  def start_link(_) do
+    Mentat.start_link(name: @cache, limit: [size: @cache_limit])
```

Comment:
This limits the cache entries to 1000 and cleans up the oldest 10% when the cache limit is reached. Probably should make this configurable later.